src/API/src/analytics/analytics.service.ts

```diff
@@ -0,0 +1,151 @@
+/* eslint-disable max-len*/
+import { Injectable, Logger } from '@nestjs/common';
+import { HttpService } from '@nestjs/axios';
+import { lastValueFrom, map } from 'rxjs';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Occurrence } from 'src/db/occurrence/entities/occurrence.entity';
+import { Repository } from 'typeorm';
+
+export const getUmamiToken = async (http: HttpService) => {
+  const user = {
+    username: 'admin',
+    password: process.env.ANALYTICS_ADMIN_PASSWORD,
+  };
+  const token = await lastValueFrom(
+    http.post(`${process.env.ANALYTICS_API_URL}/auth/login`, user).pipe(
+      map((res: any) => {
+        return res.data.token;
+      }),
+    ),
+  );
+  return token;
+};
+
+export const getWebsiteUUID = async (http: HttpService, token: string) => {
+  const websiteList = await lastValueFrom(
+    http
+      .get(`${process.env.ANALYTICS_API_URL}/websites`, {
+        headers: { Authorization: `Bearer ${token}` },
+      })
+      .pipe(
+        map((res: any) => {
+          return res.data;
+        }),
+      ),
+  );
+  const website = websiteList.find((website) => {
+    return website.name === 'vector-atlas-test';
+  });
+  const uuid = website.websiteUuid;
+  return uuid;
```

Comment:
Is this not the same as `NEXT_PUBLIC_ANALYTICS_ID`?

Reply:
It is but I realized we can't hardcode it as an env variable because my id locally will not be the same when a umami instance is spun up when deployed

Reply:
Unless that's something we can go in change manually but I am unfamiliar with deployments as a whole

Reply:
See above :-)